Swiftz/Monoidal.swift

```diff
@@ -0,0 +1,15 @@
+//
+//  Cartesian.swift
+//  Swiftz
+//
+//  Created by Robert Widmann on 5/3/16.
+//  Copyright © 2016 TypeLift. All rights reserved.
+//
+
+public protocol Cartesian : Functor {
+	associatedtype FTOP = K1<()>
+	associatedtype FTAB = K1<(A, B)>
+
+	static var unit : FTOP { get }
+	func product(r : FB) -> FTAB
```

Comment:
How deep should we go here?  Cats goes to 22, but that will balloon the implementation.